service/src/main/java/org/apache/celeborn/server/common/service/config/ConfigService.java

```diff
@@ -32,6 +32,22 @@ default DynamicConfig getTenantConfig(String tenantId) {
     }
   }
 
+  UserConfig getRawUserConfig(String tenantId, String userId);
+
+  default DynamicConfig getUserConfig(String tenantId, String userId) {
```

Comment:
If userConfig is not exist/empty, we can directly use tenantConfig instead.

```suggestion
  default DynamicConfig getTenantUserConfig(String tenantId, String userId) {
    TenantConfig tenantConfig = getRawTenantUserConfig(tenantId, userId);
    if (tenantConfig == null) {
      return getTenantConfig(tenantId);
    } else {
      return tenantConfig;
    }
  }
```

Reply:
Done